lib/blocs/root/root_bloc.dart

```diff
@@ -38,18 +39,19 @@ class RootBloc extends Bloc<RootEvent, RootState> {
         // And past state was dsiconnected.
         if (state.internetConnectionStatus == InternetConnectionStatus.disconnected) {
           emit(state.copyWith(internetConnectionStatus: InternetConnectionStatus.connected));
+          timer?.cancel();
           eventBus.fire(const ShowSnackBar('Your internet connection was restored'));
         }
       } else {
         // Network data detected but no internet connection found.
         emit(state.copyWith(internetConnectionStatus: InternetConnectionStatus.disconnected));
-        eventBus.fire(const ShowSnackBar('You are currently offline'));
+        timer = Timer(const Duration(seconds: 5), () => eventBus.fire(const ShowSnackBar('You are currently offline')));
```

Comment:
can you leave a quick comment in the code explaining why the timer is added

Reply:
done